dbms/src/DataStreams/WindowBlockInputStream.cpp

```diff
@@ -272,12 +274,123 @@ Int64 WindowTransformAction::getPartitionEndRow(size_t block_rows)
     return left;
 }
 
+RowNumber WindowTransformAction::stepForward(const RowNumber & current_row, UInt64 n)
+{
+    auto dist = distance(current_row, partition_start);
+
+    if (dist <= n)
```

Comment:
When will `dist < n`?

Reply:
> When will `dist < n`?

`select *, first_value(v) over (partition by p rows between 5 preceding and 10 following) as a from test.t;`

In this sql, n == 5 and when then distance between current_row and partition_start is lower than 5, `dist < n` will happen.